Display any non WCPay Subscriptions as manual renewal when the Subscriptions extension is deactivated

mattallan opened the pull request:
Issue: #1
P2: pdjTHR-35-p2

### Description

When the Subscriptions extension is deactivated on a store and only core is loaded via WooCommerce Payments, tokenized Subscriptions will need to be displayed as manual renewal. This is because the payment processing triggered by the action scheduler is only part of the Subscriptions extension.

As part of the issue description (#1), James gave two options for implementing this and I decided to go with adding/overriding a condition within the `$subscriptions->is_manual()` function.

This new function in core (`WC_Subscriptions_Core_Payment_Gateways::get_subscription_payment_gateway()`) makes it so we only return a valid payment gateway if the gateway is WooCommerce Payments and it's a WCPay Subscription.

If it's not, the subscription will be Manual.

### Testing Instructions

The best way to test these changes is to simulate a store deactivating Subscriptions with active subscriptions (i.e. downgrading) and checking all tokenized subscriptions (non WCPay Subscriptions) appear as manual renewal:

1. Git clone `woocommerce-subscriptions-core` into wp-plugins and checkout this branch (this will make subscriptions 4.0+ and WC payments load `subscriptions-core` from this repo)
2. First activate the Subscriptions extension (this can be any subscriptions version, i.e. 3.1.4 (before core existence), trunk (before core), or `fix/core-issue-1` branch (has core changes), it doesn’t matter - any will work)
3. Activate and enable two payment gateways; WC Payments + Stripe
4. Purchase two subscriptions, one with each payment gateway
5. With the Subscriptions extension still activated, check both payments methods are displayed under the subscriptions on the table
6. Process renewals for both, confirm renewals are still created and processed
7. Deactivate Subscriptions extension and have WooCommerce Payments and Stripe enabled.
8. Visit the subscriptions table and notice both subscriptions will appear as manual renewal
9. Purchase a WCPay Subscription and make sure it appears as WooCommerce Payments
10. Reactivate Subscriptions extension, confirm the 3 subscriptions now have their payment methods displayed.

Closes #1

Commit:
New function to only get the payemnt gateway for wcpay subscriptions

When Subscriptions extension is not active, we want any non-wcpay subscriptions to appear as manual.
This is because any tokenized subscriptions are essentially manual when using core.

Only WCPay Subscriptions should appear with a payment method set.

Commit:
Update subscription->is_manual() to return true when using tokenized subscriptions

mattallan commented:
Related PR made on the subscriptions extension: 4209-gh-woocommerce/woocommerce-subscriptions

james-allan commented:
This looks good and works well. 

The one question I had and we discussed this earlier was changing the function to be something more along the lines of: 

```php 
has_available_payment_method( $subscription ) {...} 
has_available_automatic_payment_method( $subscription ) {...}
```

I don't have a strong opinion on the final name other than it conveys that the payment method is available rather than the more general getting gateway. The addition of 'automatic' just narrows the scope even further to: does this subscription have an available payment method capable of processing automatic payments (given it is being called in the `is_manual()` function)

james-allan commented:
> The addition of 'automatic' just narrows the scope even further to: does this subscription have an available payment method capable of processing automatic payments (given it is being called in the is_manual() function)

After thinking about this some more, I don't think this is actually necessary because in the WC Subscriptions plugin the function won't be making a comment on whether the gateway is capable of processing automatic subscriptions payments, just whether it is available (active).

james-allan reviewed (changes requested):
See this comment https://github.com/Automattic/woocommerce-subscriptions-core/pull/2#issuecomment-928694446

Commit:
Refactor/rename get_subscription_payment_gateway() to has_available_payment_method()

This function doesn't need to return the payment gateway object since it's only used in a condition within is_manual()

mattallan commented:
Thanks for the review @james-allan, I've updated the function to be called `has_available_payment_method()` and refactored the function to only return a bool since we don't have a need for it returning the `WC_Payment_Gateway` object.

james-allan committed:
Fix typo in function comment

james-allan commented:
LGTM 👍
